pkg/scheduler/framework/plugins/noderesources/least_allocated_test.go

```diff
@@ -248,3 +249,64 @@ func TestNodeResourcesLeastAllocated(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeResourcesLeastAllocatedWithWeight(t *testing.T) {
+	cpuAndMemory := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("1000m"),
+						v1.ResourceMemory: resource.MustParse("2000"),
+					},
+				},
+			},
+			{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("2000m"),
+						v1.ResourceMemory: resource.MustParse("3000"),
+					},
+				},
+			},
+		},
+	}
+	tests := []struct {
+		pod          *v1.Pod
+		pods         []*v1.Pod
+		nodes        []*v1.Node
+		expectedList framework.NodeScoreList
+		name         string
+	}{
+		{
+			// CPU Score: ((4000 - 3000) *100) / 4000 = 25
+			// Memory Score: ((10000 - 5000) *100) / 10000 = 50
+			// Node1 Score: (25 * 1 + 50 * 2) / (1 + 2) = 41
+			// CPU Score: ((6000 - 3000) *100) / 6000 = 50
+			// Memory Score: ((10000 - 5000) *100) / 10000 = 50
+			// Node2 Score: (50 * 1 + 50 * 2) / (1 + 2) = 50
+			pod:          &v1.Pod{Spec: cpuAndMemory},
+			nodes:        []*v1.Node{makeNode("machine1", 4000, 10000), makeNode("machine2", 6000, 10000)},
+			expectedList: []framework.NodeScore{{Name: "machine1", Score: 41}, {Name: "machine2", Score: 50}},
+			name:         "nothing scheduled, resources requested, differently sized machines",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			snapshot := cache.NewSnapshot(test.pods, test.nodes)
+			fh, _ := framework.NewFramework(nil, nil, nil, framework.WithSnapshotSharedLister(snapshot))
+			args := &runtime.Unknown{Raw: []byte(`{"resources" : [{"name" : "memory", "weight" : 2}, {"name" : "cpu", "weight" : 1}]}`)}
```

Comment:
This should be part of the test case, but you could also just expand existing test tables.

Reply:
Done